sdk/core/azure-core/inc/http/curl/curl.hpp

```diff
@@ -374,15 +380,24 @@ namespace Azure { namespace Core { namespace Http {
      */
     CurlSession(Request& request) : m_request(request)
     {
-      this->m_pCurl = curl_easy_init();
+      this->m_connection = GetCurlConnection(this->m_request);
       this->m_bodyStartInBuffer = -1;
       this->m_innerBufferSize = Details::c_LibcurlReaderSize;
       this->m_rawResponseEOF = false;
       this->m_isChunkedResponseType = false;
       this->m_uploadedBytes = 0;
     }
 
-    ~CurlSession() override { curl_easy_cleanup(this->m_pCurl); }
+    ~CurlSession() override
+    {
+      // mark connection as reusable only if entire response was read
+      // If not, connection can't be reused because next Read will start from what it is currently
+      // in the wire. We leave the connection blocked until Server closes the connection
+      if (this->m_rawResponseEOF)
```

Comment:
Maybe we should just forcibly close the connection here now. It seems strange to me that the object is being destructed and we just let the connection linger with unread data.